internal/directives/file_deleter_test.go

```diff
@@ -0,0 +1,88 @@
+package directives
+
+import (
+	"context"
+	kargoapi "github.com/akuity/kargo/api/v1alpha1"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"os"
+	"path/filepath"
+	"testing"
+)
+
+func Test_fileDeleter_runPromotionStep(t *testing.T) {
+	tests := []struct {
+		name       string
+		setupFiles func(*testing.T) string
+		cfg        DeleteConfig
+		assertions func(*testing.T, string, PromotionStepResult, error)
+	}{
+		{
+			name: "succeeds deleting file",
+			setupFiles: func(t *testing.T) string {
+				tmpDir := t.TempDir()
+
+				path := filepath.Join(tmpDir, "input.txt")
+				require.NoError(t, os.WriteFile(path, []byte("test content"), 0o600))
+
+				return tmpDir
+			},
+			cfg: DeleteConfig{
+				Path: "input.txt",
+			},
+			assertions: func(t *testing.T, workDir string, result PromotionStepResult, err error) {
+				assert.NoError(t, err)
+				assert.Equal(t, PromotionStepResult{Status: kargoapi.PromotionPhaseSucceeded}, result)
+
+				_, statError := os.Stat("input.txt")
+				assert.ErrorIs(t, statError, os.ErrNotExist)
+			},
+		},
+		{
+			name: "succeeds deleting directory",
+			setupFiles: func(t *testing.T) string {
+				tmpDir := t.TempDir()
+				dirPath := filepath.Join(tmpDir, "dirToDelete")
+				require.NoError(t, os.Mkdir(dirPath, 0o700))
+				return tmpDir
+			},
+			cfg: DeleteConfig{
+				Path: "dirToDelete",
+			},
+			assertions: func(t *testing.T, workDir string, result PromotionStepResult, err error) {
+				assert.NoError(t, err)
+				assert.Equal(t, PromotionStepResult{Status: kargoapi.PromotionPhaseSucceeded}, result)
+
+				_, statErr := os.Stat(filepath.Join(workDir, "dirToDelete"))
+				assert.ErrorIs(t, statErr, os.ErrNotExist)
+			},
+		},
+		{
+			name: "fails for non-existent path",
```

Comment:
In addition to these tests, can you add:

- One which removes a symlink (should remove the literal symlink, and not the file).
- One which removes a file or directory within a symlink (should traverse into the symlink and remove it).

Reply:
The first test case has been added. I would like to confirm something for the second test case you'd like to be added:

The test shouldn't remove the actual symlink but only the target file(inside some other directory) pointed by a symlink directory. Correct?

Reply:
Given path `foo/file.txt`, where `foo` is a symlink to `bar/`. It should delete `bar/file.txt`.